include/eeprom/core/addresses.hpp

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include "types.hpp"
+
+namespace eeprom {
+namespace addresses {
+
+constexpr types::data_length serial_number_length = 12;
```

Comment:
Yeah I'm confused about the correspondence of this type and the PR body.

I will also note that we really do want a longer serial number. We don't get to choose this - we use what the factory wants, and what they want is a datecode of YYYYMMDD, the ordinal of when the device was made (i.e. 1 for the first device that day, 2 for the next, etc), and sometimes a letter. 12 is a good compromise, but we may even want it longer at some point, and it definitely won't be shorter.

Reply:
Nah. We'll only read / write 8 of the 12. 

That means that the python can stay as it is.

Reply:
Our current eeprom supports 8 byte reads and writes. Rather than doing multiple writes and reads to support a larger SN, which would be throw away logic, I chose to support an  8 byte SN. 

The production eeprom will support up to 64-byte reads and writes. Large enough for a complete SN.